File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/PauseIndex.py
from ravendb import StartIndexOperation, GetIndexingStatusOperation, StopIndexingOperation, StopIndexOperation
from ravendb.documents.indexes.definitions import IndexRunningStatus
from ravendb.documents.operations.definitions import VoidMaintenanceOperation

from examples_base import ExampleBase


class Foo:
    # region syntax
    class StopIndexOperation(VoidMaintenanceOperation):
        # class name has "Stop", but this is ok, this is the "Pause" operation
        def __init__(self, index_name: str): ...

    # endregion


class Pause(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_pause_index(self):
        with self.embedded_server.get_document_store("ResumeIndex") as store:
            self.add_index_orders_totals(store)
            # region pause_index
            # Define the resume index operation, pass the index name
            pause_index_op = StopIndexOperation("Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            store.maintenance.send(pause_index_op)

            # At this point:
            # Index 'Orders/Totals' is paused on the preferred node

            # Can verify the index status on the preferred node by sending GetIndexingStatusOperation
            indexing_status = store.maintenance.send(GetIndexingStatusOperation())

            index = [x for x in indexing_status.indexes if x.name == "Orders/Totals"][0]
            self.assertEqual(index.status, IndexRunningStatus.PAUSED)
            # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/PauseIndex.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use PHPUnit\Framework\TestCase;
use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexingStatus;
use RavenDB\Documents\Operations\Indexes\GetIndexingStatusOperation;
use RavenDB\Documents\Operations\Indexes\StopIndexOperation;

class PauseIndex extends TestCase
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region pause_index
            // Define the pause index operation, pass the index name
            $pauseIndexOp = new StopIndexOperation("Orders/Totals");

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($pauseIndexOp);

            // At this point:
            // Index 'Orders/Totals' is paused on the preferred node

            // Can verify the index status on the preferred node by sending GetIndexingStatusOperation
            /** @var IndexingStatus $indexingStatus */
            $indexingStatus = $store->maintenance()->send(new GetIndexingStatusOperation());

            $indexes = array_filter($indexingStatus->getIndexes()->getArrayCopy(), function ($v, $k) {
                return $v->getName() == "Orders/Totals";
            });
            /** @var IndexingStatus $index */
            $index = $indexes[0];

            $this->assertTrue($index->getStatus()->isRunning());
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax
    // class name has "Stop", but this is ok, this is the "Pause" operation
    public StopIndexOperation(?string $indexName)
    # endregion
}
*/
